pytorch_lightning/plugins/training_type/sharded.py

```diff
@@ -41,7 +46,16 @@ def _reinit_optimizers_with_oss(self):
                 optimizer = optimizer._optimizer
             if not isinstance(optimizer, OSS):
                 optim_class = type(optimizer)
-                zero_optimizer = OSS(params=optimizer.param_groups, optim=optim_class, **optimizer.defaults)
+                is_fp16 = (self.lightning_module.trainer.accelerator_connector.precision == 16)
+                zero_optimizer = OSS(
+                    params=optimizer.param_groups,
+                    optim=optim_class,
+                    # For multi-nodes, compress the model shards in fp16 before broadcasting
+                    # would improve performance. When using PyTorch AMP, it would not degrade
+                    # the model performance.
```

Comment:
```suggestion
                    # For multi-node training, compressing the model shards in fp16 before broadcasting
                    # improves performance. When using PyTorch AMP, it will not degrade
                    # the model performance.
```